ldk/node/src/hostClients/baseClient.ts

```diff
@@ -101,7 +101,7 @@ export default abstract class BaseClient<THost extends CommonHostServer>
       callback: (err: grpc.ServiceError | null, response: TResponse) => void,
     ) => void,
     builder: () => TMessage,
-    renderer: (response: TResponse) => TOutput | undefined,
+    renderer: (response: TResponse) => TOutput | PromiseLike<TOutput>,
```

Comment:
What prompted this change?

Reply:
Explained out of band, but for the record, adding `PromiseLike<TOutput>` was me not understanding the problem at all.

For everyone else's convenience:
- PR that broke this in `typescript` itself: https://github.com/microsoft/TypeScript/pull/39817
- How other big consumers are dealing with it (look at related commits): https://github.com/microsoft/vscode/issues/106358

Reply:
I've updated the PR to just not go that far ahead with the update for typescript (a patch version bump instead of minor version bump), as it really wasn't the point of this PR anyways.